### ISSUE TITLE
Local storage persistence is broken

### ISSUE THREAD

**Issue by rfvermut** [helm/chartmuseum#193]
Kubernetes persistence mounts local storage as root. And 1.8 changed docker user to non-root so chartmuseum process cannot write to /storage.

Needs init-container or something similar.

**Comment by saphoooo** [comment]
same here

**command**

```
curl --data-binary "@dummy-0.1.0.tgz" http://chartmuseum.theevilcorp.xyz/api/charts
{"error":"open /storage/dummy-0.1.0.tgz: permission denied"}
```

**logs**

```
k logs truculent-hound-chartmuseum-5d8cc546cc-q2kxm
{"L":"ERROR","T":"2018-12-14T20:25:57.839Z","M":"[3] Request served","path":"/api/charts","comment":"","latency":"4.15781ms","clientIP":"192.168.99.100","method":"POST","statusCode":500,"reqID":"9eec382b4ab55078ed786f65cc16de40"}
```

**analysis**

```
k exec -ti truculent-hound-chartmuseum-5d8cc546cc-q2kxm -- sh
/ $ cd storage/
/storage $ touch hello
touch: hello: Permission denied
/storage $ ls -l
total 0
/storage $ cd ..
/ $ ls -l
total 29744
drwxr-xr-x    2 root     root             6 Dec 14 20:25 storage
/ $ id
uid=1000(chartmuseum) gid=1000(chartmuseum)
```

**Comment by saphoooo** [comment]
In 0.7.1 chartmuseum run as root, so it works.

```
curl --data-binary "@dummy-0.1.0.tgz" http://chartmuseum.theevilcorp.xyz/api/charts
{"saved":true}
```

**Comment by jdolitsky** [comment]
@rfvermut @saphoooo @evilosa I believe that this is something that can be fixed in the chart

Using something like

```
spec:
  securityContext:
    runAsUser: 1000
    fsGroup: 1000
```

https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod

Will keep this open until that is fixed

**Comment by jdolitsky** [comment]
@rfvermut @saphoooo @evilosa 

This issue  should now be fixed in chart version 1.8.2:

```
helm repo update
helm install stable/chartmuseum --version=1.8.2
```

Please re-open this issue if any problems. Thanks!
